src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -466,9 +521,10 @@ void AssimpImporter::doOpenData(Containers::Array<char>&& data, DataFlags) {
                we didn't miss anything in the root node. */
             CORRADE_INTERNAL_ASSERT(!_importer || !root->mNumMeshes);
 
-            _f->nodes.reserve(root->mNumChildren);
-            _f->nodes.insert(_f->nodes.end(), root->mChildren, root->mChildren + root->mNumChildren);
-            _f->nodeIndices.reserve(root->mNumChildren);
+            const size_t numDescendants = countNodeDescendants(root);
+            arrayReserve(_f->nodes, numDescendants);
+            arrayAppend(_f->nodes, Containers::arrayView(const_cast<const aiNode* const*>(root->mChildren), root->mNumChildren));
```

Comment:
My idea here was that the extra `const` could hopefully avoid the need for `const_cast`? Or, if not working directly, at least it could work with `InPlaceInit`?

```suggestion
            arrayAppend(_f->nodes, InPlaceInit, root->mChildren, root->mNumChildren);
```

Reply:
It's using the [`ArrayView<const T>`](https://doc.magnum.graphics/corrade/namespaceCorrade_1_1Containers.html#a42ea2d796126a0411960911e18b35447) overload to append all children at once. The compiler doesn't know what to do when the view is non-const:
> none of the 2 overloads could convert all the argument types

And the two overloads it seems to consider are the ones taking a reference or an rvalue. I don't really know enough about overload resolution to understand why `arrayAppend(Array<T>&, ArrayView<const T>)` is not a candidate. But it's a general pain point with the function:
```c++
ArrayView<int> source = ...;
Array<int> destination;
// The cast is required here
Containers::arrayAppend(destination, Containers::arrayView<const int>(source));
```

Reply:
Ah, I guess the template type deduction simply fails there. What exactly does the `std::common_type<T>` do in the single-item overloads?

Reply:
Mhm, yes, let me try to add `std::common_type` there as well, that could hopefully fix it. Otherwise it says "deduced conflicting type for T" and fails.

Reply:
Fixed in mosra/corrade@7db68ab54666d2f83c9cc5d2775a0753677c6a09, and another annoyance in mosra/corrade@322be0685c948cc2b4a8509863925e333719fdc1. This was a massive PITA for me as well, and the fix allowed me to clean up some horrible stuff like in 373df05ffb443539ef3ff0517800cd7a863160cc. So I hope it'll work in your case as well :)

Reply:
Thanks for the fix, works great now 😌